sdk/src/instruction.rs

```diff
@@ -3,96 +3,124 @@
 use crate::{pubkey::Pubkey, short_vec, system_instruction::SystemError};
 use bincode::serialize;
 use serde::Serialize;
+use thiserror::Error;
 
 /// Reasons the runtime might have rejected an instruction.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Serialize, Deserialize, Debug, Error, PartialEq, Eq, Clone)]
 pub enum InstructionError {
     /// Deprecated! Use CustomError instead!
     /// The program instruction returned an error
+    #[error("generic instruction error")]
     GenericError,
 
     /// The arguments provided to a program instruction where invalid
+    #[error("invalid instruction argument")]
```

Comment:
Description refers to instruction but the error refers to a program getting bad arguments.  Other added `#[error`s are also doing that and probably shouldn't

Reply:
And any time the error string becomes just as complete as the comment, go ahead and delete the comment.  Don't need the maintenance burden of having to make updates in both places or confusion from them getting out of sync.

Reply:
@garious will we not lose the docs in that case?  I don't see where `thiserror` claims to generate them.  Maybe that's fine...

Reply:
Good point!  I suppose we'll need to keep the duplication until we can convince ourselves nobody reads those docs. :)

Reply:
Shouldn't take too long! :laughing: